mypy checks in pre-commit fail

## Bug description

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1
thoth/adviser/cpu_db.py:21:1: error: Library stubs not installed for "yaml" (or incompatible with Python 3.6)  [import]
thoth/adviser/prescription/v1/prescription.py:23:1: error: Library stubs not installed for "yaml" (or incompatible with Python 3.6)  [import]
thoth/adviser/pipeline_builder.py:35:1: error: Library stubs not installed for "yaml" (or incompatible with Python 3.6)  [import]
thoth/adviser/pipeline_builder.py:35:1: note: Hint: "python3 -m pip install types-PyYAML"
thoth/adviser/pipeline_builder.py:35:1: note: (or run "mypy --install-types" to install all missing stub packages)
thoth/adviser/pipeline_builder.py:35:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
thoth/adviser/predictor.py:54: error: Unused "type: ignore" comment
thoth/adviser/resolver.py:676:31: error: Returning Any from function declared to return "Union[SupportsDunderLT, SupportsDunderGT]"  [no-any-return]
thoth/adviser/resolver.py:766:[50](https://prow.operate-first.cloud/view/s3/ci-prow/prow-logs/pr-logs/pull/thoth-station_adviser/2334/pre-commit/1524015462336172032#1:build-log.txt%3A50): error: Returning Any from function declared to return "Union[SupportsDunderLT, SupportsDunderGT]"  [no-any-return]
thoth/adviser/resolver.py:1296:73: error: No overload variant of "max" matches argument types "None", "float"  [call-overload]
thoth/adviser/resolver.py:1296:73: note: Possible overload variants:
thoth/adviser/resolver.py:1296:73: note:     def [SupportsRichComparisonT] max(__arg1, SupportsRichComparisonT, SupportsRichComparisonT, *_args: SupportsRichComparisonT, key: None = ...) -> SupportsRichComparisonT
thoth/adviser/resolver.py:1296:73: note:     def [_T] max(__arg1, _T, _T, *_args: _T, key: Callable[[_T], Union[SupportsDunderLT, SupportsDunderGT]]) -> _T
thoth/adviser/resolver.py:1296:73: note:     def [SupportsRichComparisonT] max(__iterable, Iterable[SupportsRichComparisonT], *, key: None = ...) -> SupportsRichComparisonT
thoth/adviser/resolver.py:1296:73: note:     def [_T] max(__iterable, Iterable[_T], *, key: Callable[[_T], Union[SupportsDunderLT, SupportsDunderGT]]) -> _T
thoth/adviser/resolver.py:1296:73: note:     def [SupportsRichComparisonT, _T] max(__iterable, Iterable[SupportsRichComparisonT], *, key: None = ..., default: _T) -> Union[SupportsRichComparisonT, _T]
thoth/adviser/resolver.py:1296:73: note:     def [_T1, _T2] max(__iterable, Iterable[_T1], *, key: Callable[[_T1], Union[SupportsDunderLT, SupportsDunderGT]], default: _T2) -> Union[_T1, _T2]
thoth/adviser/predictors/annealing.py:69:9: error: Returning Any from function declared to return "float"  [no-any-return]
thoth/adviser/predictors/mcts.py:[65](https://prow.operate-first.cloud/view/s3/ci-prow/prow-logs/pr-logs/pull/thoth-station_adviser/2334/pre-commit/1524015462336172032#1:build-log.txt%3A65):9: error: Returning Any from function declared to return "float"  [no-any-return]
Found 9 errors in 7 files (checked 120 source files)
black....................................................................Failed
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -35,8 +35,7 @@ repos:
     hooks:
       - id: mypy
         exclude: '^(docs|tasks|tests)|setup\.py'
-        args: [--ignore-missing-imports]
-        additional_dependencies: [attrs~=19.3]
+        additional_dependencies: [attrs~=21.4, types-all]
 
   - repo: https://github.com/psf/black
     rev: 22.6.0

--- a/thoth/adviser/predictors/annealing.py
+++ b/thoth/adviser/predictors/annealing.py
@@ -21,6 +21,7 @@ from typing import Any
 from typing import List
 from typing import Optional
 from typing import Tuple
+from typing import cast
 from typing import TYPE_CHECKING
 import logging
 import random
@@ -55,7 +56,9 @@ class AdaptiveSimulatedAnnealing(Predictor):
     def _temperature_function(self, t0: float, context: Context) -> float:
         """Temperature function used to compute new temperature."""
         k = (context.accepted_final_states_count + math.log(context.iteration + 1)) / context.limit
-        temperature = t0 * self.temperature_coefficient**k
+        temperature = t0 * cast(
+            float, self.temperature_coefficient**k
+        )  # https://github.com/python/typeshed/issues/285
         _LOGGER.debug(
             "New temperature for (iteration=%d, t0=%g, accepted final states=%d, limit=%d, beam size= %d, k=%f) = %g",
             context.iteration,

--- a/thoth/adviser/predictors/mcts.py
+++ b/thoth/adviser/predictors/mcts.py
@@ -21,6 +21,7 @@ import attr
 
 from typing import Optional
 from typing import Tuple
+from typing import cast
 import logging
 import math
 import os
@@ -51,7 +52,7 @@ class MCTS(TemporalDifference):
         # This function, in comparision to TD/SA, does not need to take into account iteration as it
         # works on accepted states.
         k = context.accepted_final_states_count / context.limit
-        temperature = t0 * 0.99**k
+        temperature = t0 * cast(float, 0.99**k)  # https://github.com/python/typeshed/issues/285
         _LOGGER.debug(
             "New temperature for (iteration=%d, t0=%g, accepted final states=%d, limit=%d, beam size= %d, k=%f) = %g",
             context.iteration,

--- a/thoth/adviser/resolver.py
+++ b/thoth/adviser/resolver.py
@@ -1330,7 +1330,7 @@ class Resolver:
             self.context.accepted_final_states_count / duration,
         )
 
-        if self.beam.size > 0:
+        if max_score != float("-inf"):
             _LOGGER.info("The highest rated software stack resolved has a score of %0.2f", max_score)
 
         _LOGGER.info(

--- a/thoth/adviser/resolver.py
+++ b/thoth/adviser/resolver.py
@@ -1302,15 +1302,18 @@ class Resolver:
                     or self.context.accepted_final_states_count == 1
                 ):
                     _LOGGER.info(
-                        "Pipeline performed %d moves in the dependency graph and generated %d software stacks out "
-                        "of %d requested (pipeline pace %.02f stacks/second); top rated software stack in beam "
-                        "has a score of %.2f; top rated software stack found so far has a score of %.2f",
-                        self.context.iteration,
-                        self.context.accepted_final_states_count,
-                        self.context.limit,
-                        self.context.accepted_final_states_count / (time.monotonic() - start_time),
-                        self.beam.max().score if self.beam.size > 0 else float("nan"),
-                        float("nan") if max_score is None else max_score,
+                        f"Pipeline performed {self.context.iteration} moves in the dependency graph and generated"
+                        f"{self.context.accepted_final_states_count} software stacks out of "
+                        f"{self.context.limit} requested (pipeline pace "
+                        f"{(self.context.accepted_final_states_count / (time.monotonic() - start_time)):.02f}"
+                        f" stacks/second)"
+                        + (
+                            f";"
+                            f"top rated software stack in beam has a score of {self.beam.max().score:.2f}; "
+                            f"top rated software stack found so far has a score of {max_score:.2f}"
+                            if self.beam.size > 0
+                            else ""
+                        )
                     )
                     last_iteration_logged = self.context.iteration
 

--- a/thoth/adviser/resolver.py
+++ b/thoth/adviser/resolver.py
@@ -1302,19 +1302,16 @@ class Resolver:
                     or self.context.accepted_final_states_count == 1
                 ):
                     _LOGGER.info(
-                        f"Pipeline performed {self.context.iteration} moves in the dependency graph and generated"
-                        f"{self.context.accepted_final_states_count} software stacks out of "
-                        f"{self.context.limit} requested (pipeline pace "
-                        f"{(self.context.accepted_final_states_count / (time.monotonic() - start_time)):.02f}"
-                        f" stacks/second)"
-                        + (
-                            f";"
-                            f"top rated software stack in beam has a score of {self.beam.max().score:.2f}; "
-                            f"top rated software stack found so far has a score of {max_score:.2f}"
-                            if self.beam.size > 0
-                            else ""
-                        )
+                        "Pipeline performed %d moves in the dependency graph and generated %d software stacks out "
+                        "of %d requested (pipeline pace %.02f stacks/second)",
+                        self.context.iteration,
+                        self.context.accepted_final_states_count,
+                        self.context.limit,
+                        self.context.accepted_final_states_count / (time.monotonic() - start_time),
                     )
+                    if self.beam.size > 0:
+                        _LOGGER.info("top rated software stack in beam has a score of %.2f", self.beam.max().score)
+                    _LOGGER.info("top rated software stack found so far has a score of %.2f", max_score)
                     last_iteration_logged = self.context.iteration
 
                 yield final_state

--- a/thoth/adviser/resolver.py
+++ b/thoth/adviser/resolver.py
@@ -1333,7 +1333,7 @@ class Resolver:
             self.context.accepted_final_states_count / duration,
         )
 
-        if max_score is not None:
+        if self.beam.size > 0:
             _LOGGER.info("The highest rated software stack resolved has a score of %0.2f", max_score)
 
         _LOGGER.info(

--- a/thoth/adviser/resolver.py
+++ b/thoth/adviser/resolver.py
@@ -673,7 +673,8 @@ class Resolver:
         resolved_direct_dependencies: Dict[str, List[PackageVersion]] = self.solver.solve(
             sorted(
                 self.project.iter_dependencies(with_devel=with_devel),
-                key=lambda p: p.name,
+                key=lambda p: p.name,  # type: ignore[no-any-return]
+                # https://github.com/python/mypy/issues/9656#issuecomment-718284938
             ),
             graceful=True,
         )
@@ -763,7 +764,8 @@ class Resolver:
             for direct_dependency in package_versions:
                 self.context.register_package_version(direct_dependency)
 
-            package_versions.sort(key=lambda pv: pv.semantic_version, reverse=True)
+            package_versions.sort(key=lambda pv: pv.semantic_version, reverse=True)  # type: ignore[no-any-return]
+            # https://github.com/python/mypy/issues/9656#issuecomment-718284938
             try:
                 package_versions = list(self._run_sieves(package_versions))
             except SkipPackage as exc:
@@ -1280,7 +1282,7 @@ class Resolver:
         self.pipeline.call_pre_run()
 
         start_time = time.monotonic()
-        max_score = None
+        max_score = float("-inf")
         last_iteration_logged = 0
         try:
             for final_state in self._do_resolve_states_raw(
@@ -1293,7 +1295,7 @@ class Resolver:
                     self.context.limit,
                 )
 
-                max_score = final_state.score if max_score is None else max(max_score, final_state.score)
+                max_score = max(max_score, final_state.score)
 
                 if (
                     self.context.iteration - last_iteration_logged > self.log_iteration

--- a/thoth/adviser/resolver.py
+++ b/thoth/adviser/resolver.py
@@ -173,13 +173,13 @@ class Resolver:
         type=Optional[int],
         kw_only=True,
         default=DEFAULT_BEAM_WIDTH,
-        converter=_beam_width,  # type: ignore
+        converter=_beam_width,
     )
     limit_latest_versions = attr.ib(
         type=Optional[int],
         kw_only=True,
         default=DEFAULT_LIMIT_LATEST_VERSIONS,
-        converter=_limit_latest_versions,  # type: ignore
+        converter=_limit_latest_versions,
     )
 
     prescription = attr.ib(type=Optional["Prescription"], default=None, kw_only=True)

